### DISCUSSION TITLE
DEFI:fuzz testing:Deposit handler

### DISCUSSION THREAD

**Discussion by nikita-biichuk** [Cyfrin/foundry-full-course-cu#3100] [General]
I don't understand why we can't make this assertion:

```
assert(totalCollateralValue > totalSupply);
```

Here is the full invariant function:

```
function invariant_protocolMustHaveMoreCollateralThanTotalSupply() public view {
    uint256 totalSupply = dsc.totalSupply();
    uint256 totalWETHCollateral = IERC20(weth).balanceOf(address(dsce));
    uint256 totalWBTCCollateral = IERC20(wbtc).balanceOf(address(dsce));
    uint256 totalWETHValue = dsce.getUsdValueOfCollateral(weth, totalWETHCollateral);
    uint256 totalWBTCValue = dsce.getUsdValueOfCollateral(wbtc, totalWBTCCollateral);
    uint256 totalCollateralValue = totalWETHValue + totalWBTCValue;
    assert(totalCollateralValue > totalSupply);
}
```

The test fails with the following error:

```
[FAIL: failed to set up invariant testing environment: panic: assertion failed (0x01)] 
invariant_protocolMustHaveMoreCollateralThanTotalSupply() (runs: 0, calls: 0, reverts: 0)
```

I expected the assertion to hold because we handle the case where collateral is zero by bounding the deposit value between 1 and type(uint96).max. 

Someone kindly help

**Comment by EngrPips** [comment]
Hello @nikitabiichuk2009, Please read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/3026#discussioncomment-11650853 I believe it will help you.

**Reply by nikita-biichuk** [reply]
@EngrPips, thanks, but unfortunately it didnt help

**Reply by EngrPips** [reply]
Did you do forge clean and delete the cache folder before retrying?

**Reply by nikita-biichuk** [reply]
yeap

**Reply by EngrPips** [reply]
And you made the changes to your `Foundry.toml` file as required?

**Reply by nikita-biichuk** [reply]
Also yes

**Reply by EngrPips** [reply]
Did you save the files with `cltr +s ` or `command + s`?

**Reply by nikita-biichuk** [reply]
Of course

**Reply by EngrPips** [reply]
I am unsure what the problem is.

**Reply by EngrPips** [reply]
The suggestion in this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/3026#discussioncomment-11650853 has been the solution to this problem you are running into. I am unsure why your case is different.

**Reply by nikita-biichuk** [reply]
Yeah strange...

**Reply by EngrPips** [reply]
Indeed.
